patches/coreboot-4.8.1/0009-Add-heads-TPM-measurements-to-Skylake-Kabylake.patch

```diff
@@ -0,0 +1,181 @@
+diff --git ./src/arch/x86/acpi.c ./src/arch/x86/acpi.c
+index 8b6b2c1..fca4a76 100644
+--- ./src/arch/x86/acpi.c
++++ ./src/arch/x86/acpi.c
+@@ -48,6 +48,7 @@
+ #include <cpu/x86/lapic_def.h>
+ #include <cpu/cpu.h>
+ #include <cbfs.h>
++#include <security/tpm/tss.h>
+ 
+ u8 acpi_checksum(u8 *table, u32 length)
+ {
+@@ -1020,6 +1021,10 @@ unsigned long write_acpi_tables(unsigned long start)
+ 		return current;
+ 	}
+ 
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT)) {
++	  tlcl_measure(2, (const void*) dsdt_file, dsdt_size);
++	}
++
+ 	slic_file = cbfs_boot_map_with_leak(CONFIG_CBFS_PREFIX "/slic",
+ 				     CBFS_TYPE_RAW, &slic_size);
+ 	if (slic_file
+diff --git ./src/arch/x86/postcar.c ./src/arch/x86/postcar.c
+index 6497b73..e846b69 100644
+--- ./src/arch/x86/postcar.c
++++ ./src/arch/x86/postcar.c
+@@ -19,6 +19,7 @@
+ #include <console/console.h>
+ #include <main_decl.h>
+ #include <program_loading.h>
++#include <security/tpm/tss.h>
+ #include <soc/intel/common/util.h>
+ 
+ /*
+@@ -43,3 +44,11 @@ void main(void)
+ 	/* Load and run ramstage. */
+ 	run_ramstage();
+ }
++
++void platform_segment_loaded(uintptr_t start, size_t size, int flags)
++{
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT) && !(flags & SEG_NO_MEASURE)) {
++	  tlcl_measure(2, (const void*) start, size);
++	}
++}
++
+diff --git ./src/drivers/intel/fsp2_0/memory_init.c ./src/drivers/intel/fsp2_0/memory_init.c
+index 30987ce..124d3fa 100644
+--- ./src/drivers/intel/fsp2_0/memory_init.c
++++ ./src/drivers/intel/fsp2_0/memory_init.c
+@@ -150,10 +150,11 @@ static void do_fsp_post_memory_init(bool s3wake, uint32_t fsp_version)
+ 
+ 	/*
+ 	 * Initialize the TPM, unless the TPM was already initialized
+-	 * in verstage and used to verify romstage.
++	 * in verstage and used to verify romstage, or for measured boot.
+ 	 */
+ 	if (IS_ENABLED(CONFIG_LPC_TPM) &&
+-	    !IS_ENABLED(CONFIG_VBOOT_STARTS_IN_BOOTBLOCK))
++	    !IS_ENABLED(CONFIG_VBOOT_STARTS_IN_BOOTBLOCK) &&
++			!IS_ENABLED(CONFIG_MEASURED_BOOT))
+ 		init_tpm(s3wake);
+ }
+ 
+@@ -483,8 +484,33 @@ void fsp_memory_init(bool s3wake)
+ 	if (status != CB_SUCCESS)
+ 		die("Loading FSPM failed!\n");
+ 
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT) && IS_ENABLED(CONFIG_LPC_TPM)) {
```

Comment:
Do we need to check for CONFIG_LPC_TPM ? The Kconfig option for CONFIG_MEASURED_BOOT already depends on MAINBOARD_HAS_LPC_TPM and it does a 'select TPM' so I assume that if MEASURED_BOOT is enabled then LPC_TPM will also always be enabled.

Reply:
it's to mirror the normal call to `init_tpm`.  seems safer in case we change `CONFIG_MEASURED_BOOT` to allow for a different TPM where there is another initialization path

Reply:
Ok ,sounds good to me to keep it in that case.